src/mainwindow.cpp

```diff
@@ -3215,6 +3215,22 @@ void MainWindow::leaveEvent(QEvent *)
     this->unsetCursor();
 }
 
+/*!
+ * \brief MainWindow::changeEvent
+ */
+void MainWindow::changeEvent(QEvent *event)
+{
+#if defined(Q_OS_UNIX) && !defined(Q_OS_MACOS)
+    if (!event->spontaneous() || m_useNativeWindowFrame)
```

Comment:
What is the `spontaneous()` check meant to achieve? Why do we care whether this event "originated outside the application"?

Since this is only about adjusting to the maximized state, I think it would be appropriate to check `event->type() == QEvent::WindowStateChange` here.

Reply:
At least on Linux, this event fires *a lot* if I don't check for spontaneity. Besides, we only really care if the window was maximized/unmaximized by the window manager.

>`event->type() == QEvent::WindowStateChange`

I tried this first, but the event never fired on GNOME upon maximizing/unmaximizing... 🤔

Reply:
I guess with firing a lot, you mean without checking for `WindowStateChange`? Because `changeEvent` covers a lot of stuff, much of which is indeed not "spontaneous", but checking the type of the event should be the preferred approach.

Which event were you seeing instead on GNOME?

Reply:
Ohh, I was doing things wrong then. I was checking for `QEvent::WindowStateChange` in `MainWindow::eventFilter()`, but it never fired there.

Alright, I'll check for that in `changeEvent()`. Thanks!